app/src/lib/stores/app-store.ts

```diff
@@ -2384,16 +2385,28 @@ export class AppStore extends TypedBaseStore<IAppState> {
           // look up if the tracked branch matches a protected remote branch
           const { upstreamWithoutRemote } = gitStore.tip.branch
 
-          if (upstreamWithoutRemote !== null) {
-            const isProtected = await this.repositoriesStore.isBranchProtected(
+          // ensure the remote associated with this branch is the one we
```

Comment:
should all of this be behind the `enableBranchProtectionWarning()` flag (on line `2381`)? currently that flag is set to `development` only, so i'm pretty sure this should be in a separate conditional or that the feature flag should be changed (but i don't think i fully understand the implications of turning that feature flag on). otherwise we won't collect metrics for this outside of `development`.

Reply:
> otherwise we won't collect metrics for this outside of `development`.

I was planning to follow up with a trivial "enable this for beta" PR after this landed and before we were ready to enable this for beta, just in case it interfered with something more urgent.